flink-java/src/main/java/org/apache/flink/api/java/Utils.java

```diff
@@ -32,4 +38,44 @@ public static String getCallLocationName(int depth) {
 		}
 		return st[depth].toString();
 	}
+
+	protected static class CountHelper<T> extends RichFlatMapFunction<T, Long> {
+
+		private static final long serialVersionUID = 1L;
+
+		private final String id;
+
+		public CountHelper(String id) {
+			this.id = id;
+		}
+
+		@Override
+		public void flatMap(T value, Collector<Long> out) throws Exception {
+			getRuntimeContext().getLongCounter(id).add(1L);
```

Comment:
I think it is more efficient to have a `long`inside this function that is incremented for each record, and then aggregate this `long` to the accumulator in the `close()` method.